mongo/integration/crud_prose_test.go

```diff
@@ -113,6 +114,181 @@ func TestWriteErrorsWithLabels(t *testing.T) {
+			expectBulkError     bool
+			expectedCommandName string
+		}{
+			{
+				desc: "InsertOne schema validation errors should include Details",
+				operation: func(coll *mongo.Collection) error {
+					// Try to insert a document that doesn't contain the required properties.
+					_, err := coll.InsertOne(context.Background(), bson.D{{"nope", 1}})
+					return err
+				},
+				expectBulkError:     false,
+				expectedCommandName: "insert",
+			},
+			{
+				desc: "InsertMany schema validation errors should include Details",
+				operation: func(coll *mongo.Collection) error {
+					// Try to insert a document that doesn't contain the required properties.
+					_, err := coll.InsertMany(context.Background(), []interface{}{bson.D{{"nope", 1}}})
+					return err
+				},
+				expectBulkError:     true,
+				expectedCommandName: "insert",
+			},
+			{
+				desc: "UpdateOne schema validation errors should include Details",
+				operation: func(coll *mongo.Collection) error {
+					// Try to set "a" to be an int, which violates the string type requirement.
+					_, err := coll.UpdateOne(
+						context.Background(),
+						bson.D{},
+						bson.D{{"$set", bson.D{{"a", 1}}}})
+					return err
+				},
+				expectBulkError:     false,
+				expectedCommandName: "update",
+			},
+			{
+				desc: "UpdateMany schema validation errors should include Details",
+				operation: func(coll *mongo.Collection) error {
+					// Try to set "a" to be an int in all documents in the collection, which violates
+					// the string type requirement.
+					_, err := coll.UpdateMany(
+						context.Background(),
+						bson.D{},
+						bson.D{{"$set", bson.D{{"a", 1}}}})
+					return err
+				},
+				expectBulkError:     false,
+				expectedCommandName: "update",
+			},
+		}
+
+		for _, tc := range cases {
+			mt.RunOpts(tc.desc, validatorOpts, func(mt *mtest.T) {
+				// Insert two valid documents so that the Update* tests can try to update them.
+				{
+					_, err := mt.Coll.InsertMany(
+						context.Background(),
+						[]interface{}{
+							bson.D{{"a", "str1"}, {"b", 1}},
+							bson.D{{"a", "str2"}, {"b", 2}},
+						})
+					assert.Nil(mt, err, "unexpected error inserting valid documents: %s", err)
+				}
+
+				err := tc.operation(mt.Coll)
+				assert.NotNil(mt, err, "expected an error from calling the operation")
+				sErr := err.(mongo.ServerError)
+				assert.True(
+					mt,
+					sErr.HasErrorCode(121),
+					"expected mongo.WriteException to have error code 121 (DocumentValidationFailure)")
+
+				var details bson.Raw
+				if tc.expectBulkError {
+					bwe, ok := err.(mongo.BulkWriteException)
+					assert.True(
+						mt,
+						ok,
+						"expected error to be type mongo.BulkWriteException, got type = %T",
+						err)
+					// Assert that there is one WriteError and that the Details field is populated.
+					assert.Equal(
+						mt,
+						1,
+						len(bwe.WriteErrors),
+						"expected exactly 1 write error, but got %d",
```

Comment:
Same thing here - if we're seeing the wrong number of write errors, it'd be helpful to have the full error printed in the test failure output for debugging.

Reply:
Good idea, will do.